### PR TITLE
Fix postgres handling for unlimited varchars

### LINKED ISSUE #5238: [CT-636] [Bug] Postgres unlimited varchar default to varchar(256)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using postgres adapter with some varchar typed columns (like when using dbt_utils.type_string()), if you use incremental materialization with sync_new_columns or append_new_columns, or actually anything that will make dbt generate DDL statements, varchars max size defaults to 256 (handler in base/Column.py behaviour when char_size is None). This makes schema evolution impossible when having textual columns defined as {{ dbt_utils.type_string() }}

### Expected Behavior

The original varchar type with no limit should remain as is and not be assigned a 256 default limit, because we don't know the size limit on a varchar without explicit limits. The schema evolution should be possible, with new columns not created as varchar(256)

### Steps To Reproduce

1. Create an incremental model with on_schema_change: sync_all_columns
2. run the model for the first time
3. add a column whose type is {{ dbt_utils.type_string() }} and whose length exceeds 256 characters
4. re-run the model. It will fail because it adds a varchar(256) column

### Relevant log output

```shell
Debug logs for the schema update : 
 "database"."schema"."table":
         Schema changed: True
         Source columns not in target: [<Column col1 (character varying(256))>, <Column col2 (character varying(256))>, <Column col3 (character varying(256))>]
         Target columns not in source: [<Column col4 (character varying(256))>]
         New column types: []
(Note that all these columns are typed are varchar initially not varchar(256))
Postgres columns data check : 
```
```sql
select
          column_name,
          data_type,
          character_maximum_length,
          numeric_precision,
          numeric_scale

      from "production-data-warehouse".INFORMATION_SCHEMA.columns
      where table_name = '{{table}}'
        
        and table_schema = '{{schema}}'
        
      order by ordinal_position
```
```shell
Error: 
`value too long for type character varying(256)`
```

### Environment

```markdown
- OS: Any
- Python: 3.9-slim
- dbt: 1.0.0
```

### What database are you using dbt with?

postgres

### Additional Context

I already found out why it worked like that, and if validated will be happy to submit a PR (linked to the fact that the way postgres columns are retrieved have NULL info regarding varchar columns max_length, and thus when using it it falls back in default Column handler)
Just wondered if this shall be fixed in the PostgresColumn class (that already handles text columns this way), or if we shall just update the dbt_utils.string_type macro to use text instead of varchar for postgres

### PR DESCRIPTION
**Pull request by shrodingers**
resolves #5238 

### Description

Handles `varchar` the same way as `text` for postgres columns (since unlimited `varchar` on postgres is not implicitely limited to 256 characters as on redshift)

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by emmyoop** [comment]
Close/reopen to retrigger snyk

File: .changes/unreleased/Fixes-20220523-103843.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+kind: Fixes
+body: Remove the default 256 characters limit on postgres character varying type when
+  no limitation is set
+time: 2022-05-23T10:38:43.392232+02:00
+custom:
+  Author: shrodingers
+  Issue: "636"
```
Comment (emmyoop) [inline .changes/unreleased/Fixes-20220523-103843.yaml:7]:
```suggestion
  Issue: "5238"
```
This should be the GitHub issue number.  636 is the issue number in our Jira instance.

**Review by emmyoop — COMMENTED** [review]
Looks good but I'm going to let @McKnight-42 have the final approval.

**Review by McKnight-42 — APPROVED** [review]
@shrodingers Sorry for how long it's taken to get eyes on this, seems like a super reasonable! great conversation in the issue between you and @jtcohen6 just the minor fix @emmyoop pointed out for changelog and this should be good to merge!

**Comment by shrodingers** [comment]
Thanks @emmyoop, @McKnight-42 and @jtcohen6 ! 
Just accepted the changelog update (sorry for the late reply was on holidays)

**Comment by McKnight-42** [comment]
@shrodingers great work on this!

**Pull request merged**